cmd/ttn-lw-cli/commands/end_devices.go

```diff
@@ -331,10 +331,10 @@ var (
 
 			device.SetFields(res, "ids.dev_addr")
 			device.SetFields(res, append(append(nsPaths, asPaths...), jsPaths...)...)
-			if device.CreatedAt.IsZero() || (!res.CreatedAt.IsZero() && res.CreatedAt.Before(res.CreatedAt)) {
+			if device.CreatedAt == nil || (res.CreatedAt != nil && ttnpb.StdTime(res.CreatedAt).Before(*ttnpb.StdTime(res.CreatedAt))) { // does this make sense?
```

Comment:
No this doesn't make sense. The original check didn't make sense either; `res.CreatedAt` is checked against itself. The purpose here is to find the earliest `CreatedAt`. Also there's an obsolete parentheses group.

```suggestion
			if device.CreatedAt == nil || res.CreatedAt != nil && ttnpb.StdTime(res.CreatedAt).Before(*ttnpb.StdTime(device.CreatedAt)) {
```

Also, if you have questions, ask them in the PR comment, add a review comment, don't bury them in a code comment and/or mark the PR as draft.

Reply:
> Also there's an obsolete parentheses group

Well, relying on operator precedence for `||` and `&&` is typically considered bad practice, but okay.

> Also, if you have questions, ask them in the PR comment

Sorry, it was mind-numbing work, I forgot to get back to this.